indico/modules/users/controllers.py

```diff
@@ -177,13 +182,63 @@ class RHPersonalData(RHUserBase):
     def _process(self):
         form = UserDetailsForm(obj=FormDefaults(self.user, skip_attrs={'title'}, title=self.user._title),
                                synced_fields=self.user.synced_fields, synced_values=self.user.synced_values)
+        pic_form = UserPictureForm()
         if form.validate_on_submit():
             self.user.synced_fields = form.synced_fields
             form.populate_obj(self.user, skip=self.user.synced_fields)
             self.user.synchronize_data(refresh=True)
             flash(_('Your personal data was successfully updated.'), 'success')
             return redirect(url_for('.user_profile'))
-        return WPUser.render_template('personal_data.html', 'personal_data', user=self.user, form=form)
+        else:
+            if self.user.has_picture:
+                pic_form.picture.data = self.user
+        return WPUser.render_template('personal_data.html', 'personal_data', user=self.user, form=form,
+                                      pic_form=pic_form)
+
+
+class RHProfilePictureDisplay(RHUserBase):
+    def _process(self):
+        if not self.user.has_picture:
+            raise NotFound
+        metadata = self.user.picture_metadata
+        return send_file(metadata['filename'], BytesIO(self.user.picture), mimetype=metadata['content_type'],
+                         conditional=True)
+
+
+class RHProfilePicture(RHUserBase):
+    def _process(self):
+        f = request.files['picture']
+        try:
+            pic = Image.open(f)
+        except IOError:
+            raise UserValueError(_('You cannot upload this file as a room picture.'))
+        if pic.format.lower() not in {'jpeg', 'png'}:
```

Comment:
what about allowing gif as well?

Reply:
Would you really want people to use animated gifs as profiles pictures?

Reply:
![aAYrp7R](https://user-images.githubusercontent.com/716307/81051866-cd74b980-8ec2-11ea-9c4d-52d5b721d04a.gif)

Reply:
no but that part goes away when converting to PNG anyway